src/DataGrid.tsx

```diff
@@ -730,18 +729,18 @@ function DataGrid<R, SR, K extends Key>(
     } else if (isSamePosition(selectedPosition, position)) {
       // Avoid re-renders if the selected cell state is the same
       // TODO: replace with a #record? https://github.com/microsoft/TypeScript/issues/39831
-      scrollToCell(position);
+      scrollIntoView(gridRef.current?.querySelector('[tabindex="0"]'));
     } else {
       setSelectedPosition({ ...position, mode: 'SELECT' });
     }
   }
 
-  function scrollToCell({ idx, rowIdx }: Partial<Position>): void {
+  function scrollToColumn(idx: number): void {
```

Comment:
We may be able to render an extra div, call `scrollIntoView` and remove it from the DOM.